src/c++/perf_analyzer/data_loader.cc

```diff
@@ -343,7 +344,13 @@ DataLoader::GetInputData(
         string_data = &it->second;
         *batch1_size = string_data->size();
       }
-      *data_ptr = (const uint8_t*)&((it->second)[0]);
+
+      if (it->second.size()) {
+        *data_ptr = (const uint8_t*)&((it->second)[0]);
+      } else {
+        *data_ptr = (const uint8_t*)&(it->second);
```

Comment:
can we just set it to a null pointer here?

Reply:
What I'm trying to do is NOT have it be nullptr to differentiate between "valid but empty data" vs "invalid data".

Reply:
No, in the case that it is nullptr (the old behavior) the input is assumed to be optional and is not sent at all.  (There is different code earlier that checks for missing optional data. That passes since the data IS specified)

I've kept the code how I have it, but have filed a follow-up story to clean this up to have an explicit found/valid bool returned.